### ISSUE TITLE
Retrieve different pool types from environment

### ISSUE THREAD

**Issue by Victoremepunto** [RedHatInsights/bonfire#261]
Currently, the available `pool` [types](https://github.com/RedHatInsights/bonfire/blob/de03c7e245aa853642743ccfc9fa3e4a1cfa5df1/bonfire/config.py#L19) are hardcoded, it would be interesting to be able to fetch them from the environment of the target cluster, if possible.

**Comment by bsquizz** [comment]
I agree, this is something I've wanted to change in the near future too. It should be feasible to get the list of pools dynamically by running `oc get namespacepool` -- only other thing we should be careful about is just making sure all "normal ephemeral users" on the cluster have the k8s role bindings set up properly so that they have access to get/list namespace pools (they might already have that access, I forget)
